modulegen/main.go

```diff
@@ -170,14 +171,13 @@ func main() {
 	fmt.Println("Thanks!")
 }
 
-func generate(example Example, rootDir string) error {
+func generate(example Example, ctx *Context) error {
```

Comment:
I did not notice this: for consistency, let's pass the context as first arg.

But let's do this in a separate PR in order to merge this one as is :)